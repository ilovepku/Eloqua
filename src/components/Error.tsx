import React from 'react'
import {View, Text} from 'react-native'
import tailwind from 'tailwind-rn'

interface Props {
  errMsg: string
}
const Error: React.FC<Props> = ({errMsg}) => (
  <View style={tailwind('flex-1 bg-white items-center justify-center')}>
    <Text style={tailwind('text-lg text-red-600')}>{errMsg}</Text>
  </View>
)

export default Error
