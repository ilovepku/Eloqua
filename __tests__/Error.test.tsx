import React from 'react'
import {render} from '@testing-library/react-native'

import Error from '../src/components/error/Error'

describe('Error', () => {
  it('renders correctly', () => {
    const tree = render(<Error errMsg="mock error" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
