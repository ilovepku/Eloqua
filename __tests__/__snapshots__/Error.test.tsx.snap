// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(229, 62, 62, 1)",
        "fontSize": 18,
      }
    }
  >
    mock error
  </Text>
</View>
`;
